test/books.test.js

```diff
@@ -1,71 +1,156 @@
 const cds = require('@sap/cds');
 
+
+const test = cds.test(__dirname + '/..');
+const { POST, GET, PUT, DELETE, expect, axios } = test;
+const newBook = {
+  title: 'kozina',
+  author: 'saaid'
+};
+const newPurchaseHistory = {
+  book_id: "7812e4c4-9db5-5687-bb64-1b216bb2f569",
+  user_id: "7812e4c4-874d-2365-opl9-09876bb2f790",
+  date: "2022-12-31"
+};
+
+
+let url = (entity, id) => {
+  return id ? `/odata/v4/catalog/${entity}('${id}')` : `/odata/v4/catalog/${entity}`;
+}
+let bookTestId;
+let purchaseHistorTestId;
```

Comment:
There is no reason to make these variables global to the file.
It should be in the describe scope at least.